class.cpp

```diff
@@ -8,22 +12,128 @@ class Minefield
     bool is_q_marked[15][15];    //true if q_marked, false otherwise
     bool is_flagged[15][15];     //true if flagged, false otherwise
     short number[15][15];        //stores total number of mines in the eight surrounding tiles
-    char state[15][15];          //store info which can be revealed to player
-		// this array holds 'B' if a bomb is to be shown
-	// it holds 'O' for a open tile which has no number
-	// it holds '1', '2'... and so one for numers to be displayed
-	// it holds '?' for question mark
-	// it holds 'F' for flag
-	
+    char state[15][15];          //stores info which can be revealed to player
     bool loss;                   //becomes true when game is lost, false till then
     int num_swept;               //holds number of swept tiles
     int num_flagged;             //holds number of flags used
+    //legend
+    char flag_char = 'F';
+    char q_mark_char = '?';
+    char unswept_char = 'U';
+    char swept_char = 'S';
+    char mine_char = '*';
 	
-    void set_field(int i_ex,int j_ex)   //mutator
-    {
-        //fill this up
-        //sets both mines and numbers
+    void set_field(int i_ex,int j_ex)               //mutator
+   {  
+	
+       for (int i = 0; i < r; i++)                                      
+         {
+           for (int j = 0; j < c; j++)
+           {
+            number[i][j] = 0;
+            is_mine[i][j] = false;
+            }
+         }
+	   

```

Comment:
I think there is no need to make all elements of number to 0 and all elements of is_mine to false. Because these will be stored in them default, So we can remove lines 29-36

Reply:
Actually, Java has default values when declaring a data structure but C++ does not. Default value of boolean data type may contain garbage value in C++. So better to set all as 0.

Reply:
Sachin is right. Legal initial values do not exist in c++

Reply:
yes i tested it without the initialization to 0 and false ,it had garbage values